From 6faa09556a94c2b86a2b999eb1fde3125f17f8f3 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 28 Jan 2015 23:10:37 -0500
Subject: [infiniband] Revert: ipoib: Consolidate rtnl_lock tasks in workqueue

Message-id: <5a4804182cb98769549d822559f6817b38ae4e0b.1422486318.git.dledford@redhat.com>
Patchwork-id: 103872
O-Subject: [Patch RHEL7 8/8] Revert "[infiniband] ipoib: Consolidate rtnl_lock tasks in workqueue"
Bugzilla: 1179740
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

This reverts commit fdf575828e66146aa15aa86936afc78664583c7a.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/ulp/ipoib/ipoib_multicast.c b/drivers/infiniband/ulp/ipoib/ipoib_multicast.c
index a0a4285..d4e0057 100644
--- a/drivers/infiniband/ulp/ipoib/ipoib_multicast.c
+++ b/drivers/infiniband/ulp/ipoib/ipoib_multicast.c
@@ -190,6 +190,12 @@ static int ipoib_mcast_join_finish(struct ipoib_mcast *mcast,
   spin_unlock_irq(&priv->lock);
   priv->tx_wr.wr.ud.remote_qkey = priv->qkey;
   set_qkey = 1;
+
+  if (!ipoib_cm_admin_enabled(dev)) {
+   rtnl_lock();
+   dev_set_mtu(dev, min(priv->mcast_mtu, priv->admin_mtu));
+   rtnl_unlock();
+  }
  }
 
  if (!test_bit(IPOIB_MCAST_FLAG_SENDONLY, &mcast->flags)) {
@@ -365,8 +371,6 @@ void ipoib_mcast_carrier_on_task(struct work_struct *work)
  }
 
  rtnl_lock();
- if (!ipoib_cm_admin_enabled(priv->dev))
-  dev_set_mtu(priv->dev, min(priv->mcast_mtu, priv->admin_mtu));
  netif_carrier_on(priv->dev);
  rtnl_unlock();
 }
-- 
1.7.1